src/ext/integrations/elasticsearch.h

```diff
@@ -0,0 +1,104 @@
+#ifndef DD_INTEGRATIONS_ELASTICSEARCH_H
+#define DD_INTEGRATIONS_ELASTICSEARCH_H
+#include "configuration.h"
+#include "defered.h"
+
+#define _DD_AL_ES(class, method) \
+    DDTRACE_DEFERED_INTEGRATION_LOADER(class, method, "DDTrace\\Integrations\\ElasticSearch\\V1\\load")
+
+static inline void _dd_es_initialize_defered_integration(TSRMLS_D) {
+    if (!ddtrace_config_integration_enabled(DDTRACE_STRING_LITERAL("elasticsearch") TSRMLS_CC)) {
+        return;
+    }
+
+    static struct ddtrace_defered_integration _es[] = {
+        _DD_AL_ES("elasticsearch\\client", "__construct"),
```

Comment:
Assuming that in order to run anyone of the below client's methods you need to construct a client, would id be the same if we drop the lines below?
```        _DD_AL_ES("elasticsearch\\client", "count"),
        _DD_AL_ES("elasticsearch\\client", "delete"),
        _DD_AL_ES("elasticsearch\\client", "exists"),
        _DD_AL_ES("elasticsearch\\client", "explain"),
        _DD_AL_ES("elasticsearch\\client", "get"),
        _DD_AL_ES("elasticsearch\\client", "index"),
        _DD_AL_ES("elasticsearch\\client", "scroll"),
        _DD_AL_ES("elasticsearch\\client", "search"),
        _DD_AL_ES("elasticsearch\\client", "update"),```

Reply:
I know too little about ES workflow. But if one of the methods (including constructor) is >always< called first then it should be safe.

Also was just wondering - that now it should also be possible to have a not-traced "trigger" method. That would cause all other integrations to load

Reply:
We'll have to implement loader script deduplication (similar to require_once) to make this work exactly as stated. Where we can use "trigger" functions that are only traced for the purpose of loading actual integration